designs/degraded-nodepools.md

```diff
@@ -0,0 +1,119 @@
+# RFC: Degraded NodePool Status Condition
+
+## Overview
+
+Karpenter can launch nodes with a NodePool that will never join a cluster when a NodeClass is misconfigured.
```

Comment:
I don't think I'd frame this as a NodeClass misconfiguration. While that's a possible failure mode, it doesn't capture everything we're looking to address. I'd frame it as "registration failures which can't be well understood or predicted by Karpenter", and then give the specific examples.